### PR TITLE
Deprecate old methods

### LINKED ISSUE #654: Make many useless methods deprecated
With libzim7 and libkiwix10 we have moved a bit of logic to libzim. Libkiwix to a certain extend is simpler and now many metods are simple wrappers over one libzim method. Beside keeping API compatibility, this as no value and ultimatively such methods will disappear. The sooner the better. We should mark these methods as deprecated and print warnings if they are used to inform users and push developers to update their code.

### PR DESCRIPTION
**Pull request by mgautierfr**
Fix #654

### CODE REVIEW

File: include/entry.h [hunk 1]
```diff
@@ -176,6 +176,10 @@ class Entry
 
   private:
     zim::Entry entry;
+
+  private:
+    Entry(zim::Entry entry, bool marker);
```
Comment (veloman-yunkan) [inline include/entry.h:181]: Explain in a comment the purpose of this constructor

**Review by veloman-yunkan — CHANGES_REQUESTED** [review]
Just one comment about a missing comment :)

File: include/entry.h (hunk 1 above)
Comment (mgautierfr) [reply include/entry.h:187]: Done

**Review by veloman-yunkan — APPROVED** [review]

**Comment by veloman-yunkan** [comment]
Must be rebased/fixed-up before merge

**Pull request merged**